From: Denis Auroux <auroux@users.sourceforge.net>
Date: Sun, 6 Sep 2020 07:30:49 -0400
Subject: upstream text box background

Text edition box background color matches paper if page has a solid
background (feature-request #17)

Upstream: 57bce1f
---
 ChangeLog      | 1 +
 src/xo-paint.c | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/ChangeLog b/ChangeLog
index d7dc86c..53e29d8 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -5,6 +5,7 @@ This version:
       wacom driver fix is committed).
   - added Russian translation (Igor Nedoboy)
   - selection resize snaps to preserve aspect ratio
+  - text edition box background color matches paper if page has a solid background
 
 Version 0.4.8.2016 (July 20, 2017 bugfix release):
  * Bug fixes:
diff --git a/src/xo-paint.c b/src/xo-paint.c
index cf9614b..9d17731 100644
--- a/src/xo-paint.c
+++ b/src/xo-paint.c
@@ -629,6 +629,10 @@ void start_text(GdkEvent *event, struct Item *item)
   gtk_widget_modify_font(item->widget, font_desc);
   rgb_to_gdkcolor(item->brush.color_rgba, &color);
   gtk_widget_modify_text(item->widget, GTK_STATE_NORMAL, &color);
+  if (ui.cur_page->bg->type == BG_SOLID) {
+    rgb_to_gdkcolor(ui.cur_page->bg->color_rgba, &color);
+    gtk_widget_modify_base(item->widget, GTK_STATE_NORMAL, &color);
+  }
   pango_font_description_free(font_desc);
 
   canvas_item = gnome_canvas_item_new(ui.cur_layer->group,
